### DISCUSSION TITLE
LoraWan node.saveSession() - EEPROM live cycle?

### DISCUSSION THREAD

**Discussion by michapr** [jgromes/RadioLib#963] [General]
Hi,
as far as I understand after every node.sendReceive the session / counter will be saved into EEPROM.
If take 6 times / hour sending a sensor value to TTN we will have 52560 sendings - so many EEPROM write cycles.

There will be more than one write process to EEPROM (I think so, because of some values will be saved) - so how long we can write to the EEPROM?

Or do I understand this saveSession not right and I'm wrong?

Thanks!
Michael

**Comment by StevenCellist** [comment]
Hi @michapr, I've started working on a wiki page that will answer these details. In short: writes only happen when the data differs from what's already in there, which means that in most cases only the FCntUp will differ - which cycles through 50 different positions, prolonging the EEPROM life cycle by 50 times ;)
If you were to save 6 times an hour, in a decent ADR environment and no frequent confirmed uplinks, you should be able to run for 50 positions * 10K write cycles / 6 uplinks per hour  / 24 hours per day = 9.5 years!
If ADR or downlinks in general happen more than once every 50 uplinks, this lifetime would decrease, but under normal circumstances 1/50 should definitely be achievable.

**Comment by michapr** [comment]
Thank you for your answer, I'm happy that you thought about this detail :)
